src/engine/AddCombinedRowToTable.h

```diff
@@ -178,12 +224,13 @@ class AddCombinedRowToIdTable {
     flushBeforeInputChange();
     mergeVocab(inputLeft, currentVocabs_.at(0));
     // The right input will be empty, but with the correct number of columns.
-    inputLeftAndRight_ =
-        std::array{detail::toView(inputLeft),
-                   IdTableView<0>{resultTable_.numColumns() -
-                                      detail::toView(inputLeft).numColumns() +
-                                      numJoinColumns_,
-                                  ad_utility::makeUnlimitedAllocator<Id>()}};
+    inputLeftAndRight_ = std::array{
+        detail::toView(inputLeft),
+        IdTableView<0>{
+            resultTable_.numColumns() +
+                static_cast<size_t>(!keepJoinColumns_) * numJoinColumns_ -
+                detail::toView(inputLeft).numColumns() + numJoinColumns_,
+            ad_utility::makeUnlimitedAllocator<Id>()}};
```

Comment:
Since this is only a dummy placeholder, we could even insert a "no-op" allocator here, to ensure it is unused. (I was thinking about one with memory limit 0).